src/test/tools/extents/extents.c

```diff
@@ -72,12 +73,33 @@ main(int argc, char *argv[])
 	if (ret)
 		goto exit_free;
 
-	unsigned e;
-	for (e = 0; e < exts->extents_count; e++)
-		printf("%lu %lu\n",
+	if (argc == 2) {
+		/* print out all extents */
+		for (unsigned e = 0; e < exts->extents_count; e++) {
 			/* extents are in bytes, convert them to sectors */
-			B2SEC(exts->extents[e].offset_physical),
-			B2SEC(exts->extents[e].length));
+			printf("%lu %lu\n",
+				B2SEC(exts->extents[e].offset_physical),
+				B2SEC(exts->extents[e].length));
+		}
+	} else {
+		/* print the physical block of a given logical one */
+		long unsigned block = (long unsigned)atol(argv[2]);
```

Comment:
IMHO: Instead of "atol" You could used "strtoul" -  "argv[2]" is guarantee to be null terminated. 
You avoid casting and could add error handling if value will be out of the range.

Reply:
Will fix in https://github.com/pmem/pmdk/pull/2936